src/IO/ReadWriteBufferFromHTTP.cpp

```diff
@@ -545,6 +548,9 @@ bool ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::nextImpl()
             if (e.code() == POCO_EMFILE)
                 throw;
 
+            if (e.code() == ErrorCodes::TOO_MANY_REDIRECTS)
+                throw;
+
```

Comment:
```suggestion
            if (e.code() == POCO_EMFILE || e.code() == ErrorCodes::TOO_MANY_REDIRECTS)
                throw;
```

And we can update the comment above this check `/// Too many open files - non-retryable.` -> `/// Too many open files or redirects - non-retryable.` (github doesn't allow me to change this line in suggestion)